data/util/NumericShaperUtility.js

```diff
@@ -0,0 +1,294 @@
+					                 ["\uFDFD", "\uFDFD"],
+					                 ["\uFEFF", "\uFEFF"]];
+						
+					/* As of Unicode 7.0, the latin script is contained in the following blocks:
+					 * 
+					 * Basic Latin, 0000–007F.
+					 * Latin Extended-A, 0100–017F
+					 * Latin Extended-B, 0180–024F
+					 * IPA Extensions, 0250–02AF
+					 * Spacing Modifier Letters, 02B0–02FF
+					 * Phonetic Extensions, 1D00–1D7F 
+					 * Phonetic Extensions Supplement, 1D80–1DBF
+					 * Latin Extended Additional, 1E00–1EFF
+					 * Superscripts and Subscripts, 2070-209F
+					 * Letter-like Symbols, 2100–214F
+					 * Number Forms, 2150–218F
+					 * Latin Extended-C, 2C60–2C7F
+					 * LatinExtended-D, A720–A7FF
+					 * Latin Extended-E, AB30–AB6F
+					 * Alphabetic Presentation Forms (Latin ligatures) FB00–FB4F
+					 * Halfwidth and Fullwidth Forms (fullwidthLatin letters) FF00–FFEF
+					 */
+					var weakLatin = [["\u0000", "\u0040"],
+					                 ["\u005B", "\u0060"],
+					                 ["\u007B", "\u007F"],
+					                 ["\u0080", "\u00A9"],
+					                 ["\u00AB", "\u00B4"],
+					                 ["\u00B6", "\u00B9"],
+					                 ["\u00BB", "\u00BF"],
+					                 ["\u00D7", "\u00D7"],
+					                 ["\u00F7", "\u00F7"],
+					                 ["\u02B9", "\u02BA"],
+					                 ["\u02C2", "\u02CF"],
+					                 ["\u02D2", "\u02DF"],
+					                 ["\u02E5", "\u02ED"],
+					                 ["\u02EF", "\u02FF"],
+					                 ["\u2070", "\u2070"],
+					                 ["\u2074", "\u207E"],
+					                 ["\u2080", "\u208E"],
+					                 ["\u2100", "\u2101"], 
+					                 ["\u2103", "\u2106"], 
+					                 ["\u2108", "\u2109"], 
+					                 ["\u2114", "\u2114"], 
+					                 ["\u2116", "\u2118"], 
+					                 ["\u211E", "\u2123"], 
+					                 ["\u2125", "\u2125"], 
+					                 ["\u2127", "\u2127"], 
+					                 ["\u2129", "\u2129"], 
+					                 ["\u212E", "\u212E"], 
+					                 ["\u213A", "\u213B"], 
+					                 ["\u2140", "\u2144"], 
+					                 ["\u214A", "\u214D"],
+					                 ["\u2150", "\u215F"], 
+					                 ["\u2189", "\u2189"],
+					                 ["\uA720", "\uA721"],
+					                 ["\uA788", "\uA788"],
+					                 ["\uFF01", "\uFF20"], 
+					                 ["\uFF3B", "\uFF40"], 
+					                 ["\uFF5B", "\uFF65"], 
+					                 ["\uFFE0", "\uFFE6"], 
+					                 ["\uFFE8", "\uFFEE"]];
+					
+					if(this._binarySearch(strongArabic, char) > -1){
+						this._context = 2;
+					}else if(this._binarySearch(weakArabic, char) > -1 || this._binarySearch(weakLatin, char) > -1){
+						return;
+					}else{
+						this._context = 1;
+					}
+				},
+				
+				_binarySearch: function(/*Array*/ arr, /*string*/ key) {
```

Comment:
The lo-dash example isn't really fair, because `String#indexOf()` doesn't have a flag to specify that the array is sorted.    But you did make me interested enough to test this myself, and as I suspected, the regexp version is faster, actually from 10-100 times faster.   See http://bill.dojotoolkit.org/strong.html.

BTW, note that the regexp is structurally simple, essentially /[a-zA-Z]/ but just with more character ranges.